cmd/data.go

```diff
@@ -102,14 +104,23 @@ func (cmd *DataCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface
 		cmd.filePrefix = dbName + "."
 	}
 
-	conv := internal.MakeConv()
-	err = conversion.ReadSessionFile(conv, cmd.sessionJSON)
+	client, err := utils.GetClient(ctx, dbURI)
 	if err != nil {
-		return subcommands.ExitUsageError
+		err = fmt.Errorf("can't create client for db %s: %v", dbURI, err)
+		return subcommands.ExitFailure
 	}
-	if targetProfile.TargetDb != "" && conv.TargetDb != targetProfile.TargetDb {
-		err = fmt.Errorf("running data migration for Spanner dialect: %v, whereas schema mapping was done for dialect: %v", targetProfile.TargetDb, conv.TargetDb)
-		return subcommands.ExitUsageError
+	defer client.Close()
+
+	// For csv mode, skip loading a session file.
+	if sourceProfile.Driver != constants.CSV {
```

Comment:
we should avoid mentioning of csv or any specific profile here.
Similar to UseTargetSchema, we can add a method - supportSessionFile or something on the sourceProfile

Reply:
Yes we should, my bad I missed this

Reply:
Using UseTargetSchema as well for this differentiation since they pretty much are the same functionally.